heron/packing/src/java/org/apache/heron/packing/roundrobin/RoundRobinPacking.java

```diff
@@ -402,11 +410,10 @@ public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> c
     int initialNumInstance = TopologyUtils.getTotalInstance(topology);
     double initialNumInstancePerContainer = (double) initialNumInstance / initialNumContainer;
 
-    Map<String, Integer> currentComponentParallelism = currentPackingPlan.getComponentCounts();
     Map<String, Integer> newComponentParallelism =
         getNewComponentParallelism(currentPackingPlan, componentChanges);
 
-    int newNumInstance = TopologyUtils.getTotalInstance(currentComponentParallelism);
+    int newNumInstance = TopologyUtils.getTotalInstance(newComponentParallelism);
```

Comment:
this is a fix of some issue?

Reply:
Yes, this is exactly where the problem is. Previously, it is wrongfully using the old plan's component parallelism to calculate the new number of containers, but that's always going to return the same number of containers as the old plan...

Reply:
KK.